### ISSUE TITLE
Doing the validation in the middle of the training

### ISSUE THREAD

**Issue by tchoi1991** [MIC-DKFZ/nnUNet#1344]
Hello!
Thank you for your work and I'm glad that I can use nnUNet!

But I have a question about the validation!
I forgot to edit the epoch to 500. So I wanted to stop training and validate that fold.

For that, I used,

nnUNet_train 3d_fullres 900 1 -val

it's was not working with some error message.

Is that possible to do validation in the middle of the training?
or do I have to finish training anyway?

**Comment by FabianIsensee** [comment]
Hey, training needs to be finished for you to be able to run the validation. If you terminate early then the learning rate schedule will not have had a chance to complete. If you want to train for fewer epoch nnU-Net needs to know this from the start (via a custom trainer)

**Comment by tchoi1991** [comment]
Thank you so much!

**Comment by tomato18463** [comment]
In my case I found my model was overfitting after a few epochs, so I did break the training as an early stop. Doing more epochs won't get things better anyway. I find what caused the error when doing validation with early stop is that the code can't find checkpoint_final.pth. Can I just save my last checkpoint as checkpoint_final.pth to get the validation work? Will it cause any problems or erroneous results (e.g. some varibles can only be obtained at the end of the training, and their intemediate values at the middle of training is unreliable)?
